### DISCUSSION TITLE
How to train a model across multiple datasets to predict sales at many stores?

### DISCUSSION THREAD

**Discussion by tszumowski** [ourownstory/neural_prophet#535] [Q&A - forecasting best practices] [answered]
I understand from the README that it supports multiple covariates. And I understand I can make multiple models, one for each time series, from another GitHub issue here. Does it support multiple time series datasets, also known as multi-instance?

As a more concrete example, consider this [OJ Sales Dataset](https://docs.microsoft.com/en-us/azure/open-datasets/dataset-oj-sales-simulated?tabs=azureml-opendatasets). In this case, the data contains weekly sales of orange juice over 121 weeks. There are 3,991 stores included and three brands of orange juice per store so that 11,973 models can be trained.
I understand one can train independent models for each of the stores. However, I was interested in knowing if the package can take in data from multiple stores to learn correlations between them.

**Comment by mavarick** [comment]
According my experiences, the answer is no.

**Comment by ourownstory** [comment, marked as answer]
# Answer to: "Does NeuralProphet support multiple series datasets?"

@tszumowski 
# Yes
If you are referring to 'global modelling' aka shared model weights across all stores.
The feature is not well documented yet, but should become clearer end of week. (See answer in #531)

# No
If you are looking to do 'VAR' - explicitly correlating all stores and all products (likely not what you plan to do).

# Yes
If you are looking for a way to regress one store's product sales onto the same or another stores product sales.

# Here's what I would do:
1) Create 3 dataset collections, with one product as the target series each. 
Within each collection, create one dataset for each store. 

- Add the store's other two items as lagged regressors (additional columns in dataframes)
- Add the 1-10 closest stores' same item sales as additional lagged regressors (additional columns in dataframes)

2) Train a global model for each of your 3 products. (I'd use local normalization once #507 is merged).
Note: the input will need to be a dictionary containing each dataset as an entry.

3) Now you can use the 3 models to predict all products at all stores. 
Note: the input will need to be a dictionary containing each dataset as an entry.

Did I understand your question and task right?

Please feel free to reach out to me on Slack if I do not answer fast enough here:
https://join.slack.com/t/neuralprophet/shared_invite/zt-sgme2rw3-3dCH3YJ_wgg01IXHoYaeCg
